File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.java
class Solution {
    public String reverseVowels(String s) {
        char[] vow = new char[s.length()];
        char[] cons =s.toCharArray();
        int j=0;
        for(int i=0;i<s.length();i++)
        {
            char ch=s.charAt(i);
            if(ch=='a' || ch=='A' || ch=='e' || ch=='E' || ch=='i' || ch=='I' || ch=='o' || ch=='O' || ch=='u' || ch=='U')
            {
                vow[j]=ch;
                j++;
            }
        }
        j--;
        for(int i=0;i<s.length();i++)
        {
            int ch=s.charAt(i);
            if(ch=='a' || ch=='A' || ch=='e' || ch=='E' || ch=='i' || ch=='I' || ch=='o' || ch=='O' || ch=='u' || ch=='U')
            {
                cons[i]=vow[j];
                j--;
            }
        }
        return new String(cons);
    }
}

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        vow=""
        for i in s:
            if i in 'aeiouAEIOU':
                vow+=i
        j=len(vow)-1
        result=""
        for i in range(0,len(s)):
            if s[i] in 'aeiouAEIOU':
                result+=vow[j]
                j-=1
            else:
                result+=s[i]
        return result

        